src/App/NHL/PlayersList/PlayerCrudModal/PlayerCrudModal.jsx

```diff
@@ -1,23 +1,82 @@
 import React, { useEffect, useState } from "react";
-import { Button, Image, Modal, Form, Icon, Input } from "semantic-ui-react";
+import {
+    Button,
+    Image,
+    Modal,
+    Form,
+    Icon,
+    Placeholder,
+    Header,
+    Checkbox,
+    Label,
+} from "semantic-ui-react";
+import { useFormik } from "formik";
+
 import { myFetch } from "../../../../utils/myFetch";
-import s from "./PlayerCrudModal.less";
+import { getDifference } from "../../../../utils/getDifference";
 import { positions } from "../../../../utils/constants";
 
+const initial = {
+    firstName: "",
+    middleName: "",
+    lastName: "",
+    birthDate: "",
+    team: "",
+    position: "",
+    id: "",
+    forceRefresh: false,
+};
+
+const validate = (values) => {
+    const errors = {};
+    const regExpDate = /\d{4}\-(0?[1-9]|1[012])\-(0?[1-9]|[12][0-9]|3[01])*/;
+
+    if (!values.firstName) {
+        errors.firstName = "Required";
+    } else if (values.firstName.length > 20) {
+        errors.firstName = "Must be 20 characters or less";
+    }
+
+    if (values.middleName && values.middleName.length > 20) {
+        errors.middleName = "Must be 20 characters or less";
+    }
+
+    if (!values.lastName) {
+        errors.lastName = "Required";
+    } else if (values.lastName.length > 20) {
+        errors.lastName = "Must be 20 characters or less";
+    }
+
+    if (values.birthDate && !regExpDate.test(String(values.birthDate))) {
+        errors.birthDate = "Invalid date";
+    }
+
+    return errors;
+};
+
 export const PlayerCrudModal = ({ open, onClose, value }) => {
     const [player, setPlayer] = useState({});
-    const [edit, setEdit] = useState(false);
-    const [difference, setDifference] = useState({});
-    const [formValid, setFormValid] = useState(false);
-    const [errors, setErrors] = useState({
-        firstName: "",
-        middleName: "",
-        lastName: "",
-    });
-    const [dirties, setDirties] = useState({
-        firstName: false,
-        middleName: false,
-        lastName: false,
+    const [playerData, setPlayerData] = useState(initial);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(false);
```

Comment:
Try to combine loading, error, and player to one state, like const [playerFrame, setPlayerFrame] = useState({ player: undefined, loading: true, error: false })